libs/types/src/locations.rs

```diff
@@ -14,21 +14,20 @@ use cfg_primitives::AccountId;
 use frame_support::RuntimeDebugNoBound;
 use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
-use sp_core::{crypto::AccountId32, H256};
-use sp_runtime::traits::{BlakeTwo256, Hash};
-use staging_xcm::VersionedLocation;
+use sp_core::crypto::AccountId32;
+// Please note that if this version change,
+// a migration could be required in those places where
+// RestrictedTransferLocation is stored
+use staging_xcm::v4::Location;
 
 use crate::domain_address::DomainAddress;
 /// Location types for destinations that can receive restricted transfers
 #[derive(Clone, RuntimeDebugNoBound, Encode, Decode, Eq, PartialEq, MaxEncodedLen, TypeInfo)]
 pub enum RestrictedTransferLocation {
 	/// Local chain account sending destination.
 	Local(AccountId),
-	/// XCM MultiLocation sending destinations.
-	/// Using hash value here as Multilocation is large -- v1 is 512 bytes, but
-	/// next largest is only 40 bytes other values aren't hashed as we have
-	/// blake2 hashing on storage map keys, and we don't want the extra overhead
-	XCM(H256),
+	/// XCM Location sending destinations.
+	Xcm(Location),
```

Comment:
@onnovisser from this, you do not need to create the hash, just put the location (will be probably a `VersionedLocation`)